builds/e2e/templates/e2e-run.yaml

```diff
@@ -65,8 +70,6 @@ steps:
     E2E_REGISTRIES__0__PASSWORD: $(TestContainerRegistryPassword)
     E2E_ROOT_CA_PASSWORD: $(TestRootCaPassword)
     E2E_BLOB_STORE_SAS: $(TestBlobStoreSas)
-    http_proxy: $(Agent.ProxyUrl)
-    https_proxy: $(Agent.ProxyUrl)
```

Comment:
Where did this configuration move to?

Reply:
It was simply removed because the end-to-end tests ran just fine behind a proxy with it. I believe the agent machines already have the proxy variables set (system wide) in the environment.

Reply:
How did those agent machines get the proxy variables set? Do we have a specific 1ES agent built for this proxy test with env vars?

Reply:
The proxy tests are still using custom agents, not 1ES. And looking at the [deployment script](https://github.com/Azure/iotedge/blob/release/1.2/builds/e2e/proxy/configure_runner.sh), I don't think the proxy vars are set system-wide. Stay tuned, I need to investigate this a little more...

Reply:
OK, figured it out. Basically everywhere in the end-to-end tests, we use the proxy URL value from context.json, we don't rely on the environment. That's why removing the env vars from the YAML file doesn't hurt anything...we aren't relying on those values anyway. In fact, if we _were_ relying on them, we would have exposed a bug in `e2e.yaml` for all the non-proxy jobs. That is, when $(Agent.ProxyUrl) is empty, the value of `https_proxy` in the "Run tests" task is the string literal "$(Agent.ProxyUrl)". 😲

As it turns out, the `IoTEdge_check` test is not proxy-aware. You can see this by looking at the output of the test in the proxy job of the latest end-to-end run:

```
‼ aziot-edge package is up-to-date - Warning
    Error while fetching latest versions of edge components: could not send HTTP request
    caused by: error trying to connect: Connection timed out (os error 110)
```

Notice `iotedge check` considers this a warning not an error, which means it won't fail the test. So that's another reason why this has gone unnoticed...

Because the test isn't proxy-aware, it _would_ rely on `https_proxy` in the environment, except that it executes `iotedge check` as sudo, which uses a different environment. Running as sudo is completely unnecessary because the tests already run as sudo (needed to install edge packages), so in my PR I removed sudo from the command. With my changes, the `IoTEdge_check` test suddenly picked up the value of `https_proxy` from the environment, and started failing in the non-proxy jobs. Removing the env vars corrected the problem.

Now I need to see about getting this check to _pass_ in the proxy job. I'll need to turn on verbose logging for the proxy job, as we discussed, to confirm.

Reply:
Update: Getting the "aziot-edge package is up-to-date" to pass in the tests will take a little more work, and is outside the scope of this PR. In a nutshell, `iotedge check` is inconsistent about where it looks for the value of `https_proxy`, and the "aziot-edge package is up-to-date" check, in particular, is looking in the wrong place. I've filed a separate bug to track that issue. For purposes of this PR, I confirmed that the `IoTEdge_check` test is passing.